content/en/service_management/app_builder/embedded_apps.md

```diff
@@ -28,21 +28,74 @@ The App Editor modal appears, allowing you to select an app and provide it with
+
+
+### Template variable examples
+
+To populated a select component with a list of all available template variables, add the following template expression to your select component's **Options** field:
+
+{{< code-block lang="json" disable_copy="false">}}
+${global?.dashboard?.templateVariables?.map(tvar => tvar.name )}
 {{< /code-block >}}
 
-### Scope your app dynamically
+To list all of the available values of a specific template variable, use the following template expression:
 
-App elements that are linked with dashboard template variables update in tandem with the values of the template variables on the dashboard. For example, selecting a particular `instance_id` value through the template variable dropdown or directly from a graph adds the `instance_id` value to the app's filter as well, so that you can take any necessary actions on that instance:
+{{< code-block lang="json" disable_copy="false">}}
+${global?.dashboard?.templateVariables?.find(v => v.name === '<TEMPLATE_VARIABLE_NAME>')?.availableValues}
+{{< /code-block >}}
 
-{{< img src="service_management/app_builder/embedded_apps/template_variables.mp4" alt="Selecting a template variable value from a graph" video="true">}}
+To get the selected value of a template variable, use the following template expressions:
+
+- For a single-select template variable:
+   {{< code-block lang="json" disable_copy="false">}}
+${global?.dashboard?.templateVariables?.find(v => v.name === '<TEMPLATE_VARIABLE_NAME>')?.value}
+{{< /code-block >}}
+- For a multi-select template variable:
+   {{< code-block lang="json" disable_copy="false">}}
+${global?.dashboard?.templateVariables?.find(v => v.name === '<TEMPLATE_VARIABLE_NAME>')?.values}
+{{< /code-block >}}
+
+### Time frame examples
+
+To get the time frame start value, use the following template expressions:
+
+- For the numerical timestamp:
+   {{< code-block lang="json" disable_copy="false">}}
+${global?.dashboard?.timeframe?.start}
+{{< /code-block >}}
+- For a formatted date and time:
+   {{< code-block lang="json" disable_copy="false">}}
+${new Date(global?.dashboard?.timeframe?.start).toLocaleString()}
+{{< /code-block >}}
+
+To get the time frame end value, use the following template expressions:
+
+- For the numerical timestamp:
+   {{< code-block lang="json" disable_copy="false">}}
+${global?.dashboard?.timeframe?.end}
+{{< /code-block >}}
+- For a formatted date and time:
+   {{< code-block lang="json" disable_copy="false">}}
+${new Date(global?.dashboard?.timeframe?.end).toLocaleString()}
+{{< /code-block >}}
+
+To add a button that sets the value of a date range picker component to the time frame of the dashboard, perform the following steps:
+
+1. Add a date range picker component to your app and name it "dateRangePicker0".
+1. Add a button to your app. Under **Events**, fill in the following values:
```

Comment:
```suggestion
1. Add a button to your app.
1. Under **Events**, fill in the following values:
```
I think this could be split into separate steps for clarity? Feel free to ignore at your discretion based on how it looks in the UI.